sk_sidekick/lib/src/commands/release_command.dart

```diff
@@ -0,0 +1,355 @@
+import 'package:pub_semver/pub_semver.dart';
+import 'package:sidekick_core/sidekick_core.dart';
+import 'package:sk_sidekick/sk_sidekick.dart';
+import 'package:yaml/yaml.dart';
+
+class ReleaseCommand extends Command {
+  @override
+  final String description = 'Releases new version of a package';
+
+  @override
+  final String name = 'release';
+
+  @override
+  String get invocation => super.invocation.replaceFirst(
+        '[arguments]',
+        '[package-path]',
+      );
+
+  @override
+  Future<void> run() async {
+    final package = DartPackage.fromArgResults(argResults!);
+
+    print('Hey ${_getGitUserName() ?? 'developer'}, '
+        'you started the release process for package:${package.name}.');
+    final proceed = confirm(
+      'Do you want to release a new version?',
+      defaultValue: false,
+    );
+    if (!proceed) {
+      exitCode = 1;
+      return;
+    } else {
+      print(' ');
+    }
+
+    _warnIfNotOnDefaultBranch(package.root);
+
+    while (_gitRepoHasChangesIn(package.root)) {
+      ask(
+        red('package:${package.name} contains local changes. '
+            'Please remove/commit them, then hit enter to continue'),
+        required: false,
+      );
+    }
+
+    final nextReleaseChangelog = _prepareNextReleaseChangelog(package);
+
+    // this needs to be executed before _bumpVersion, otherwise package.version already returns the next version
+    final currentPackageVersionTag = '${package.name}-v${package.version}';
+    final nextVersion = await _bumpVersion(package);
+    final nextPackageVersionTag = '${package.name}-v$nextVersion';
+
+    print('Creating changelog ...');
+    final changelog = package.root.file('CHANGELOG.md');
+    final now = DateTime.now();
+    final date = '${now.year}-${now.month}-${now.day}';
+    changelog.writeAsStringSync('''
+# Changelog
+
+## [$nextVersion](https://github.com/phntmxyz/sidekick/compare/$currentPackageVersionTag..$nextPackageVersionTag) ($date)
+
+${nextReleaseChangelog.trim()}
+
+${changelog.readAsStringSync().replaceFirst('# Changelog', '').trimLeft()}''');
+
+    final bool lock = package == skProject.sidekickPackage;
+    if (lock) {
+      print('Locking dependencies ...');
+      await runSk(['lock-dependencies', package.root.path]);
+    }
+
+    final tag = '${package.name}-v$nextVersion';
+    final commitMessage = 'Prepare release $tag';
+    print('Creating commit "Prepare release "$commitMessage" '
+        'and tagging release as $tag');
+    for (final cmd in [
+      'git add -A ${package.root.path}',
+      'git commit -m "$commitMessage"',
+      'git tag $tag'
+    ]) {
+      cmd.start(workingDirectory: repository.root.path);
+    }
```

Comment:
When releasing `sidekick`, and the dependencies get locked, we should not commit all changes on `main`.
Changelog and version bump can go to main, but locking + tagging should happen on a separate branch

Reply:
locking + tagging is now done on `${package.name}-release` branch